File: app/src/main/java/com/damianperon/memorygamekotlin/adapters/CardsAD.java
package com.damianperon.memorygamekotlin.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.recyclerview.widget.RecyclerView;

import com.damianperon.memorygamekotlin.R;
import com.wajahatkarim3.easyflipview.EasyFlipView;

import org.jetbrains.annotations.NotNull;

import java.util.List;

public class CardsAD extends RecyclerView.Adapter<CardsAD.ViewHolder> {

    private int[] mData;
    private LayoutInflater mInflater;
    private ItemClickListener mClickListener;

    // data is passed into the constructor
    public CardsAD(Context context, int[] data, ItemClickListener mClickListener) {
        this.mInflater = LayoutInflater.from(context);
        this.mData = data;
        this.mClickListener = mClickListener;
    }

    // inflates the row layout from xml when needed
    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = mInflater.inflate(R.layout.card, parent, false);
        return new ViewHolder(view);
    }

    // binds the data to the TextView in each row
    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        Integer resource = mData[position];
        holder.front.setImageResource(resource);
       //holder.myEasyFlipView.flipTheView();
    }

    // total number of rows
    @Override
    public int getItemCount() {
        return mData.length;
    }


    // stores and recycles views as they are scrolled off screen
    public class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
        EasyFlipView myEasyFlipView;
        ImageView front = itemView.findViewById(R.id.cardfront);


        ViewHolder(View itemView) {
            super(itemView);
            myEasyFlipView = itemView.findViewById(R.id.flippingcard);
            itemView.setOnClickListener(this);
        }

        @Override
        public void onClick(View view) {
            myEasyFlipView.flipTheView();
            if (mClickListener != null) {
                mClickListener.onItemClick(myEasyFlipView, mData[getAdapterPosition()]);
            }
        }
    }

    // convenience method for getting data at click position
    Integer getItem(int id) {
        return mData[id];
    }

    // allows clicks events to be caught
    void setClickListener(ItemClickListener itemClickListener) {
        this.mClickListener = itemClickListener;
    }

    // parent activity will implement this method to respond to click events
    public interface ItemClickListener {
        void onItemClick(EasyFlipView view, int position);
    }
}

File: app/src/main/java/com/damianperon/memorygamekotlin/adapters/CardsAD.kt
package com.damianperon.memorygamekotlin.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView
import com.damianperon.memorygamekotlin.R
import com.wajahatkarim3.easyflipview.EasyFlipView

class CardsAD(
    context: Context?,
    data: IntArray,
    mClickListener: ItemClickListener?
) : RecyclerView.Adapter<CardsAD.ViewHolder>() {
    private val mData: IntArray
    private val mInflater: LayoutInflater
    private var mClickListener: ItemClickListener?
    // inflates the row layout from xml when needed
    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): ViewHolder {
        val view = mInflater.inflate(R.layout.card, parent, false)
        return ViewHolder(view)
    }

    // binds the data to the TextView in each row
    override fun onBindViewHolder(
        holder: ViewHolder,
        position: Int
    ) {
        val resource = mData[position]
        holder.front.setImageResource(resource)
        //holder.myEasyFlipView.flipTheView();
    }

    // total number of rows
    override fun getItemCount(): Int {
        return mData.size
    }

    // stores and recycles views as they are scrolled off screen
    inner class ViewHolder internal constructor(itemView: View) :
        RecyclerView.ViewHolder(itemView), View.OnClickListener {
        var myEasyFlipView: EasyFlipView
        var front =
            itemView.findViewById<ImageView>(R.id.cardfront)

        override fun onClick(view: View) {
            myEasyFlipView.flipTheView()
            if (mClickListener != null) {
                mClickListener!!.onItemClick(myEasyFlipView, mData[adapterPosition])
            }
        }

        init {
            myEasyFlipView = itemView.findViewById(R.id.flippingcard)
            itemView.setOnClickListener(this)
        }
    }

    // convenience method for getting data at click position
    fun getItem(id: Int): Int {
        return mData[id]
    }

    // allows clicks events to be caught
    fun setClickListener(itemClickListener: ItemClickListener?) {
        mClickListener = itemClickListener
    }

    // parent activity will implement this method to respond to click events
    interface ItemClickListener {
        fun onItemClick(view: EasyFlipView?, position: Int)
    }

    // data is passed into the constructor
    init {
        mInflater = LayoutInflater.from(context)
        mData = data
        this.mClickListener = mClickListener
    }
}